Sage/carbon
# test: update jest config to ignore lib/ and esm/

### Proposed behaviour

- Add list of regex patterns to `jest.conf.js` to ignore tests located production build directories `/lib` and `/esm`.

### Current behaviour

- Jest manages to pick up `__spec__.js` files in the  `/lib` and `/esm`.

### Checklist




### Testing instructions

- Verify any tests in `node_modules` are still ignored by jest.
- Verify no other tests apart from those in `lib` and `esm` are ignored.
